pkg/resources/kubespan/peer_spec.go

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package kubespan
+
+import (
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"inet.af/netaddr"
+)
+
+// PeerSpecType is type of PeerSpec resource.
+const PeerSpecType = resource.Type("KubeSpanPeerSpecs.kubespan.talos.dev")
+
+// PeerSpec is produced from cluster.Affiliate which has KubeSpan information attached.
+//
+// PeerSpec is identified by the public key.
+type PeerSpec struct {
+	md   resource.Metadata
+	spec PeerSpecSpec
+}
+
+// PeerSpecSpec describes PeerSpec state.
+type PeerSpecSpec struct {
+	Address             netaddr.IP         `yaml:"address"`
+	AdditionalAddresses []netaddr.IPPrefix `yaml:"additionalAddresses"`
+	Endpoints           []netaddr.IPPort   `yaml:"endpoints"`
```

Comment:
I'm a bit torn here about the IPPort as oppose of a "host" port (`net.UDPAddress`).  There is the option/possibility of DNS names being used, which case they should be resolved at _use_ time, rather than at spec time.  In fact, the current controller (and discovery service) allows DNS names, and these are further populated from the `v1.Node` addresses, which include DNS names.

I have, at various times, used each, and I just don't know whether it is worth the extra complexity to retain DNS resolution or not.

Reply:
good point, let's keep it in the back in our brains as we move through the implementation. I feel DNS is tricky a bit, as it might resolve to multiple IPs and we should try them all, so ultimately Wireguard input should be IP addresses? but we certainly might need more than a single address here.

I also think once we publish `ExternalIPs` as `Addresses` internally, so that node is aware of the external IPs, they will get pushed to the discovery as well. So my point - we can have it all, it's more about the addresses the node has (even if the address is not assigned to any interface at the moment)

Reply:
It's more than "should", it's "must".  The input to wireguard itself must be an IP address, not a DNS name.  And yes, DNS is absolutely a tricky bit.  For instance, there is also a SRV record system for Wireguard.  Should we support that, too?  Should be try append domains on top of the hostname?  All of them?  Intermediaries?

So basically, I basically agree with this approach for now.  DNS is just too sticky to dive into right now.  I think we can probably extend this resource later, if we need to, by having names as a separate field or struct or whatever.

Reply:
What I'm trying to say is that Wireguard endpoint is ultimately `IP:port` pair, so the input should be exactly that, while the code building the list of the endpoints might use DNS to resolve DNS names into IP list and populate the endpoints properly.